File: src/main/java/aubay/lu/projetrh/service/UtilisateurService.java
package aubay.lu.projetrh.service;

import aubay.lu.projetrh.model.Utilisateur;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface UtilisateurService {

    public boolean isUserExisting(String userLogin);

    public boolean mailAddressInUse(String mailAddress);

    public List<Utilisateur> getAllUtilisateurs();

    public Optional<Utilisateur> getUtilisateurById(UUID userid);

    public Optional<Utilisateur> getUtilisateurByLogin(String userlogin);

    public Optional<Utilisateur> getCandidatById(UUID candidatId);

    public List<Utilisateur> getAllCandidats();

    public Utilisateur createUtilisateur(Utilisateur utilisateur);

    public Utilisateur createCandidat(Utilisateur utilisateur);

    public Utilisateur updateUtilisateur(Utilisateur utilisateur);

    public Utilisateur updateCandidat(Utilisateur utilisateur);

    public String deleteUtilisateur(UUID userid);

    public Utilisateur userProfileUpdate(Utilisateur utilisateur, String userlogin);

    public Utilisateur userSelfRetrieve(String userlogin);

    public void deleteCandidat(UUID candidatId);

}

File: src/main/resources/projetrh/src/app/_services/_restricted/utilisateur.service.ts
import {Injectable} from "@angular/core";
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {lastValueFrom, map, Observable, switchMap} from "rxjs";
import {Utilisateur} from "../../_models/utilisateur.model";


@Injectable({
  providedIn: 'root'
})

export class UtilisateurService{

  utilisateursList!: Utilisateur[]

  constructor(private http: HttpClient){}


  async getAllUtilisateur(): Promise<Utilisateur[]>{
    let response = await lastValueFrom(this.http.get<Utilisateur[]>('http://localhost:8080/admin/utilisateur/all'))
    this.utilisateursList = response
    return this.utilisateursList
  }

  async getAllCandidats(): Promise<Utilisateur[]> {
    let response = await lastValueFrom(this.http.get<Utilisateur[]>('http://localhost:8080/moderateur/utilisateur/candidats'))
    this.utilisateursList = response
    return this.utilisateursList
  }

  getUtilisateurById(userId: any): Observable<Utilisateur>{
    return this.http.get<Utilisateur>(`http://localhost:8080/admin/utilisateur/id/${encodeURIComponent(userId)}`)
  }

  getUtilisateurByLogin(userLogin: string): Observable<Utilisateur>{
    return this.http.get<Utilisateur>(`http://localhost:8080/admin/utilisateur/login/${userLogin}`)
  }

  createUtilisateur(data: any, headers : {headers: HttpHeaders}): Observable<any> {
    return this.http.post('http://localhost:8080/admin/utilisateur/create', data, headers)
  }

  createCandidat(data: any, headers : { headers: HttpHeaders }): Observable<any>{
    return this.http.post('http://localhost:8080/moderateur/utilisateur/candidat/create', data, headers)
  }

  updateUtilisateur(utilisateur: Utilisateur): Observable<Utilisateur> {
    return this.http.put<Utilisateur>('http://localhost:8080/admin/utilisateur/update', utilisateur)
  }

  deleteUtilisateur(userId: String): void {
    this.http.delete<Utilisateur>(`http://localhost:8080/admin/utilisateur/delete/${userId}`)
  }

}
